lxd/storage/common.go

```diff
@@ -0,0 +1,114 @@
+package storage
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"syscall"
+
+	"github.com/lxc/lxd/lxd/state"
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/api"
+)
+
+type driverCommon struct {
+	s *state.State
+
+	poolID int64
+	pool   *api.StoragePool
+
+	volume *api.StorageVolume
+
+	sTypeVersion string
+}
+
+func (d *driverCommon) commonInit(s *state.State, pool *api.StoragePool, poolID int64, volume *api.StorageVolume) {
+	d.s = s
+	d.pool = pool
+	d.poolID = poolID
+	d.volume = volume
+}
+
+func (d *driverCommon) GetVersion() string {
+	return d.sTypeVersion
+}
+
+func (d *driverCommon) UsesThinpool() bool {
```

Comment:
@monstermunchkin this seems like something specific to LVM, is it really shared for all storage engines?

Reply:
It is specific to LVM, yes. However, I thought it'd be cleaner to add it to the common code instead of all other drivers just returning `false` in this function given that no LVM-specific code is called.

Reply:
@monstermunchkin I see, in the current code base I can only see an unexported `usesThinpool` (https://github.com/lxc/lxd/search?q=usesThinpool&unscoped_q=usesThinpool) used only by LVM, so maybe this doesn't need to be part of the interface at all, and can just be part of the unexported implementation of LVM?

Reply:
I just checked in my other branch, and the function is used in `lxd/storage.go` when creating a container. Maybe we can work around this, and remove the function from the interface.

Reply:
@monstermunchkin yes I think that would be good, especially as in current master its only used in LVM related places. Do you remember why you used it in lxd/storage.go in your other branch?